includes/Media.php

```diff
@@ -273,4 +277,45 @@ public static function upload_mimes( array $mime_types ) {
 		$mime_types['svg'] = 'image/svg+xml';
 		return $mime_types;
 	}
+
+	/**
+	 * Deletes associated poster image when a video is deleted.
+	 *
+	 * This prevents the poster image from becoming an orphan because it is not
+	 * displayed anywhere in WordPress or the story editor.
+	 *
+	 * @param int $attachment_id ID of the attachment to be deleted.
+	 *
+	 * @return void
+	 */
+	public static function delete_video_poster( $attachment_id ) {
+		remove_action( 'pre_get_posts', [ __CLASS__, 'filter_poster_attachments' ] );
+		$query = new WP_Query(
+			[
+				'fields'         => 'ids',
+				'post_status'    => 'any',
+				'post_type'      => 'attachment',
+				'post_parent'    => $attachment_id,
+				// phpcs:disable WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+				'meta_query'     => [
+					[
+						'key'     => self::POSTER_POST_META_KEY,
+						'compare' => 'EXISTS',
+					],
+				],
+				'no_found_rows'  => true,
+				'posts_per_page' => 1,
```

Comment:
Is there a change that they might be more than 1 here?

Reply:
If we don‘t do a meta query, yes.

Reply:
I really think we should limit to 10 and just when we loop through check if the meta exists at the point. There is a possibility that something went wrong and you have two uploaded poster as browser was closed before a REST API completes request or something. Unlikely but possible.